File: src/main/java/models/Product.java

package models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.j256.ormlite.field.DatabaseField;

import java.math.BigDecimal;

public class Product {

    @SerializedName("id")
    @Expose
    @DatabaseField(id = true)
    private int id;
    @SerializedName("name")
    @Expose
    @DatabaseField(columnName = "name")
    private String name;
    @SerializedName("short_name")
    @Expose
    @DatabaseField
    private String shortName;
    @SerializedName("unit")
    @Expose
    @DatabaseField
    private String unit;
    @SerializedName("integer")
    @Expose
    @DatabaseField
    private boolean integer;
    @SerializedName("barcode")
    @Expose
    @DatabaseField(columnName = "barcode")
    private String barcode;
    @SerializedName("vendor")
    @Expose
    @DatabaseField
    private String vendor;
    @SerializedName("producer")
    @Expose
    @DatabaseField
    private String producer;
    @SerializedName("price")
    @Expose
    @DatabaseField
    private Double price;
    @SerializedName("count")
    @Expose
    @DatabaseField
    private String count;
    @SerializedName("reserved")
    @Expose
    @DatabaseField
    private String reserved;
    @SerializedName("min_count")
    @Expose
    @DatabaseField
    private Double minCount;

    public Product(String name, Double price) {
        name = this.name;
        price = this.price;
    }

    public Product() {

    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getShortName() {
        return shortName;
    }

    public void setShortName(String shortName) {
        this.shortName = shortName;
    }

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    public boolean isInteger() {
        return integer;
    }

    public void setInteger(boolean integer) {
        this.integer = integer;
    }

    public String getBarcode() {
        return barcode;
    }

    public void setBarcode(String barcode) {
        this.barcode = barcode;
    }

    public String getVendor() {
        return vendor;
    }

    public void setVendor(String vendor) {
        this.vendor = vendor;
    }

    public String getProducer() {
        return producer;
    }

    public void setProducer(String producer) {
        this.producer = producer;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getCount() {
        return count;
    }

    public void setCount(String count) {
        this.count = count;
    }

    public String getReserved() {
        return reserved;
    }

    public void setReserved(String reserved) {
        this.reserved = reserved;
    }

    public Double getMinCount() {
        return minCount;
    }

    public String getMinCountFormat() {
        return String.format("%.2f", minCount);
    }

    public void setMinCount(Double minCount) {
        this.minCount = minCount;
    }

    //format-----------------------------------------------------------------------------
    public String getPriceFormat(){
        return String.format("%.2f", price);
    }
    //-----------------------------------------------------------------------------------
}

File: src/main/java/models/Product.kt
package models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.j256.ormlite.field.DatabaseField

class Product(@SerializedName("id")
              @Expose(serialize = false)
              @DatabaseField(id = true)
              var id: Int = 0,

              @SerializedName("name")
              @Expose
              @DatabaseField(columnName = "name")
              var name: String,

              @SerializedName("short_name")
              @Expose
              @DatabaseField
              var shortName: String,

              @SerializedName("unit")
              @Expose
              @DatabaseField
              var unit: String,

              @SerializedName("integer")
              @Expose
              @DatabaseField
              var isInteger: Boolean,

              @SerializedName("barcode")
              @Expose
              @DatabaseField(columnName = "barcode")
              var barcode: String,

              @SerializedName("vendor")
              @Expose
              @DatabaseField
              var vendor: String,

              @SerializedName("producer")
              @Expose
              @DatabaseField
              var producer: String,

              @SerializedName("price")
              @Expose
              @DatabaseField
              var price: Double,

              @SerializedName("count")
              @Expose
              @DatabaseField
              var count: Double,

              @SerializedName("reserved")
              @Expose
              @DatabaseField
              var reserved: Double,

              @SerializedName("min_count")
              @Expose
              @DatabaseField
              var minCount: Double,

              @SerializedName("parent")
              @Expose
              @DatabaseField(columnName = "sub_category")
              var parent: Int) {

    constructor() : this(0, "", "", "", false, "", "", "", 0.0, 0.0, 0.0, 0.0, 0)

    private fun str(d: Double) =
            if (isInteger) {
                d.toInt().toString()
            } else {
                String.format("%.2f", d)
            }

    fun getStrCount() = str(count)
    fun getStrReserved() = str(reserved)
    fun getStrMinCount() = str(minCount)
    fun getPriceFormat() = String.format("%.2f", price)
}
